File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/asn1/ansi/x957/DssSigValue.java
package aladdin.asn1.ansi.x957; 
import aladdin.asn1.*; 
import aladdin.asn1.Integer; 
import java.io.*; 

// DssSigValue ::= SEQUENCE {
//		r            INTEGER,
//		s            INTEGER
// }

public final class DssSigValue extends Sequence<Integer>
{
    private static final long serialVersionUID = -8928732534194593289L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(Integer.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(Integer.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public DssSigValue(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public DssSigValue(Integer r, Integer s) { super(info, r, s); }

	public final Integer r() { return get(0); }
	public final Integer s() { return get(1); }
}

File: Source/Aladdin.CAPI.ANSI/ASN1/ANSI/X957/DssSigValue.cs
using System;
using System.Runtime.Serialization;

// DssSigValue ::= SEQUENCE {
//		r            INTEGER,
//		s            INTEGER
// }

namespace Aladdin.ASN1.ANSI.X957
{
	[Serializable]
	public class DssSigValue : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<Integer>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<Integer>().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected DssSigValue(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public DssSigValue(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public DssSigValue(Integer r, Integer s) : base(info, r, s) {}

		public Integer R { get { return (Integer)this[0]; } }
		public Integer S { get { return (Integer)this[1]; } }
	}
}
